### PR TITLE
Bm notification type icons

### PR DESCRIPTION
**Pull request by bamandel**
Used dynamically generated icons for Notification types. We still need the notification type legend as the audit table needs this.

### CODE REVIEW

File: alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderDistributionUIConfig.java [hunk 1]
```diff
@@ -72,15 +72,17 @@ public ProviderDistributionUIConfig(final String label, final String urlName, fi
         final ConfigField notificationTypesField = SelectConfigField.createRequired(KEY_NOTIFICATION_TYPES, LABEL_NOTIFICATION_TYPES, DESCRIPTION_NOTIFICATION_TYPES, false, true,
             providerContent.getContentTypes()
                 .stream()
-                .map(ProviderContentType::getNotificationType)
-                .map(LabelValueSelectOption::new)
+                .map(providerContentType -> new LabelValueSelectOption(
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderDistributionUIConfig.java]: Can we make this a private method and just use the method reference here?

File: alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderDistributionUIConfig.java [hunk 2]
```diff
@@ -72,15 +72,17 @@ public ProviderDistributionUIConfig(final String label, final String urlName, fi
...
+                    providerContentType.getNotificationType(),
+                    providerContentType.getFontAwesomeIcon()))
                 .sorted()
                 .collect(Collectors.toList()));
         final ConfigField formatField = SelectConfigField.createRequired(KEY_FORMAT_TYPE, LABEL_FORMAT, DESCRIPTION_FORMAT,
             providerContent.getSupportedContentFormats()
                 .stream()
                 .map(FormatType::name)
-                .map(LabelValueSelectOption::new)
+                .map(name -> new LabelValueSelectOption(WordUtils.capitalizeFully(name.replace("_", " ")), name))
```
Comment (gkillough) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderDistributionUIConfig.java]: Method reference? Whenever I see nested functions inside of a Stream.map(), I question the complexity.

File: src/main/java/com/synopsys/integration/alert/provider/blackduck/descriptor/BlackDuckContent.java [hunk 3]
```diff
@@ -118,7 +118,9 @@
         List.of(createStringField(createJsonPath(JsonField.FORMAT_DOUBLE_REPLACEMENT, JSON_FIELD_CONTENT, JSON_FIELD_BOM_COMPONENT), JSON_FIELD_BOM_COMPONENT, FieldContentIdentifier.TOPIC, LABEL_PROJECT_NAME),
             createStringField(createJsonPath(JsonField.FORMAT_DOUBLE_REPLACEMENT, JSON_FIELD_CONTENT, JSON_FIELD_BOM_COMPONENT), JSON_FIELD_BOM_COMPONENT, FieldContentIdentifier.SUB_TOPIC, LABEL_PROJECT_VERSION_NAME),
             createStringField(createJsonPath(JsonField.FORMAT_DOUBLE_REPLACEMENT, JSON_FIELD_CONTENT, JSON_FIELD_BOM_COMPONENT), JSON_FIELD_BOM_COMPONENT, FieldContentIdentifier.CATEGORY_ITEM, LABEL_BOM_COMPONENT)
-        ));
+        ),
+        "user-edit"
```
Comment (gkillough) [inline src/main/java/com/synopsys/integration/alert/provider/blackduck/descriptor/BlackDuckContent.java]: Can we make these constants? The random String is hard to follow with all that is going on in the rest of the initialization.

File: src/main/java/com/synopsys/integration/alert/provider/blackduck/descriptor/BlackDuckContent.java [hunk 4]
```diff
@@ -156,7 +159,8 @@
             createObjectField(createJsonPath(JsonField.FORMAT_DOUBLE_REPLACEMENT, JSON_FIELD_CONTENT, JSON_FIELD_POLICY_INFOS), JSON_FIELD_POLICY_INFOS, FieldContentIdentifier.CATEGORY_ITEM, LABEL_POLICY_INFO_LIST,
                 new TypeRef<List<PolicyInfo>>() {})
 
-        )
+        ),
+        "exclamation-circle"
```
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/alert/provider/blackduck/descriptor/BlackDuckContent.java]: Shouldn't this be the same icon as a policy violation just a different color?

File: alert-common/src/main/java/com/synopsys/integration/alert/common/provider/ProviderContentType.java [hunk 5]
```diff
@@ -33,17 +33,28 @@
 
 public class ProviderContentType extends Stringable {
     private final String notificationType;
+    private String fontAwesomeIcon;
     private final Collection<JsonField<?>> notificationFields;
 
-    public ProviderContentType(@NotNull final String notificationType, @NotNull final Collection<JsonField<?>> notificationFields) {
+    // TODO temporarily pass in the icon until we refactor ContentTypes and allow people to create this mapping elsewhere
+    public ProviderContentType(@NotNull final String notificationType, @NotNull final Collection<JsonField<?>> notificationFields, String fontAwesomeIcon) {
```
Comment (psantos1113) [inline alert-common/src/main/java/com/synopsys/integration/alert/common/provider/ProviderContentType.java]: Just formatting here.  Remove the final on the constructors
